rct229/rules/section22/section22rule16.py

```diff
@@ -0,0 +1,105 @@
+from rct229.rule_engine.rule_base import RuleDefinitionBase
+from rct229.rule_engine.rule_list_indexed_base import RuleDefinitionListIndexedBase
+from rct229.rule_engine.user_baseline_proposed_vals import UserBaselineProposedVals
+from rct229.schema.config import ureg
+from rct229.utils.jsonpath_utils import (
+    find_all,
+    find_exactly_one_with_field_value,
+)
+from rct229.utils.pint_utils import CalcQ
+from rct229.utils.std_comparisons import std_equal
+
+APPLICABLE_SYS_TYPES = [
+    "SYS-7",
+    "SYS-8",
+    "SYS-11.1",
+    "SYS-11.2",
+    "SYS-12",
+    "SYS-13",
+    "SYS-7B",
+    "SYS-8B",
+    "SYS-11B",
+    "SYS-12B",
+    "SYS-13B",
+]
+REQUIRED_LOW_DESIGN_WETBULB_TEMP = ureg("55 degF")
+REQUIRED_HIGH_DESIGN_WETBULB_TEMP = ureg("90 degF")
+
+
+class Section22Rule16(RuleDefinitionListIndexedBase):
+    """Rule 16 of ASHRAE 90.1-2019 Appendix G Section 22 (Chilled water loop)"""
+
+    def __init__(self):
+        super(Section22Rule16, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, True, False),
+            each_rule=Section22Rule16.HeatRejectionRule(),
+            index_rmr="baseline",
+            id="22-16",
+            description="The baseline condenser-water design supply temperature shall be calculated using the cooling tower approach to the 0.4% evaporation design wet-bulb temperature, valid for wet-bulbs from 55°F to 90°F.",
+            rmr_context="ruleset_model_instances/0",
+            list_path="heat_rejections[*]",
+        )
+
+    def is_applicable(self, context, data=None):
+        rmi_b = context.baseline
+        # FIXME: replace with baseline_system_types = get_baseline_system_types(rmi_b) when get_baseline_system_types
+        #  is ready.
+        baseline_system_types = {
+            "SYS-7": ["hvac_sys_7"],
+            "SYS-11B": ["hvac_sys_11_b"],
+        }
+        # if any system type found in the APPLICABLE_SYS_TYPES and wet-bulb temp is in the correct range then return applicable.
+        return any(
+            [key in APPLICABLE_SYS_TYPES for key in baseline_system_types.keys()]
+        ) and any(
+            REQUIRED_LOW_DESIGN_WETBULB_TEMP.to(ureg.kelvin)
+            <= heat_rejection_wetbulb_temp_b.to(ureg.kelvin)
+            <= REQUIRED_HIGH_DESIGN_WETBULB_TEMP.to(ureg.kelvin)
+            for heat_rejection_wetbulb_temp_b in find_all(
+                "heat_rejections[*].design_wetbulb_temperature", rmi_b
+            )
+        )
+
+    def create_data(self, context, data):
+        rmi_b = context.baseline
+        supply_temperature_dict_b = {
+            heat_rejection_loop: find_exactly_one_with_field_value(
```

Comment:
As discussed yesterday, this portion needs to be updated.

Reply:
Fixed. Thank you Weili!